windows/inc/leatherman/windows/user.hpp

```diff
@@ -0,0 +1,30 @@
+/**
+ * @file
+ * Declares utility functions for querying user properties
+ */
+#pragma once
+
+#include <string>
+
+namespace leatherman { namespace windows { namespace user {
+
+    /**
+     * Determines whether the current process has Administrator privileges and can be expected to succeed at
+     * tasks restricted to Administrators.
+     * @return True if the current process has Administrator privileges, otherwise false.
+     */
+    bool is_admin();
+
+    /**
+     * Query token membership to determine whether the current user is a member of the Administrators group.
+     * @return True if user is an Administrator, otherwise false.
+     */
+    bool check_token_membership();
```

Comment:
Today I only see this used in the `is_admin` implementation. Is it of general utility? If not, should we not expose it? 

Either way, is there a better name for the function? Like `in_admin_group`? Otoh if `check_token_membership` will be intuitively obvious to someone doing windows work, then just ignore me :)